tests/spawner_test.py

```diff
@@ -2,13 +2,27 @@
 
 from __future__ import annotations
 
+import asyncio
+from datetime import timedelta
+
 import pytest
 
+from rsp_restspawner.errors import SpawnFailedError
 from rsp_restspawner.spawner import LabStatus, RSPRestSpawner
 
 from .support.controller import MockLabController
 
 
+async def gather_progress(
```

Comment:
I find the overloading of "gather" a little confusing (yes, even though this is only ever used in a `gather()` call).  I would have used `collect_progress` I think.

Reply:
Good idea. Done.